packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -166,9 +166,11 @@ local function fetch_and_store_marathon_apps(auth_token)
           end
        end
 
-       -- next() returns nil if table is empty.
-       local i, task = next(tasks)
-       if i == nil then
+       -- Randomly select the task.
```

Comment:
@stg- I just realised something - the task you pick here will be used by ALL requests until the next cache refresh, i.e. the task AR is proxing requests to will change very 25s, during that period all the requests will go to the same task.

If we want to really load-balance the requests, then we first need to store the routing data for all the task with given ApplicationID, and then we should be picking randomly one in the `/service` endpoint routing code[1].

[1] https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/lib/service.lua#L83-L108